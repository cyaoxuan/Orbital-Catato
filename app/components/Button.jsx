import { View } from "react-native";
import { Button } from "react-native-paper";

const PillButton = (props) => {
    return (
        <View style={{ flexDirection: "row" }}>
                <Button style = {{ margin: 10, width: props.width }}
                    disabled={props.disabled}
                    mode={props.mode}
                    onPress={props.onPress}>
                        {props.label}
                </Button>
        </View>
    )
}

export { PillButton };