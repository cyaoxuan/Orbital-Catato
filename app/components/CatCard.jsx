import { Card } from "react-native-paper";
import { IconTextField } from "./InfoText";

const CatCard = (props) => {
    return (
        <Card style={{ height: 280, width: 280, margin: 4 }} 
            onPress={() => {}}
            mode="elevated">
            <Card.Cover style={{ height: 140, width: 280, resizeMode: "cover"}} 
                source={props.catImage} 
            />
            <Card.Title title={props.catName} titleVariant="headlineMedium"/>
            <Card.Content style={{ paddingBottom: 0, paddingHorizontal: 4 }}>
                <IconTextField 
                    iconName={props.iconName1} 
                    iconSize={24} 
                    variant="titleLarge"
                    field={props.field1}
                    info={props.info1}
                />
                <IconTextField 
                    iconName={props.iconName2} 
                    iconSize={24} 
                    variant="titleLarge" 
                    field={props.field2}
                    info={props.info2}
                />
            </Card.Content>
        </Card>
    );
};

export { CatCard };