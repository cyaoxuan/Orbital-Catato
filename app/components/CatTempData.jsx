// Temp data to showcase static pages

const catViewProps = [{
    iconName1: "information-circle",
    field1: "Status: ",
    iconName2: "location",
    field2: "Last Seen: "
}, {
    iconName1: "time",
    field1: "Last Fed: ",
    iconName2: "location",
    field2: "Last Seen: "
}];

const concernCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Engineering"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Skitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "New",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Utown"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Mitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Missing",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Science"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Bitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Arts"
}];

const fedCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[1].iconName1,
    field1: catViewProps[1].field1,
    info1: "19/05 12:01",
    iconName2: catViewProps[1].iconName2,
    field2: catViewProps[1].field2,
    info2: "Engineering"
}];

export { concernCatCards, fedCatCards }