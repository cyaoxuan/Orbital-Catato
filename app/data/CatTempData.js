// Temp data to showcase static pages
const cats = [{
    catID: 1,
    name: "Kitty",
    photoURLs: [require("../../assets/temp-cat.jpg"), require("../../assets/catato-logo.png")],
    gender: "M",
    birthYear: 2019,
    sterilised: true,
    keyFeatures: "Grey tabby and loves to sleep on his back",
    lastSeenLocation: "Engineering", // Will eventually be a geohash
    lastSeenTime: new Date(2023, 4, 19, 14, 21, 0, 0), // Will be converted from firebase timestamp
    lastFedTime: new Date(2023, 4, 19, 12, 1, 0, 0), 
    concernStatus: ["Injured"], 
    concernDesc: "Seen limping, possibly right front leg injured"
}, {
    catID: 2,
    name: "Skitty",
    photoURLs: [require("../../assets/temp-cat.jpg")],
    gender: null,
    birthYear: null,
    sterilised: false,
    keyFeatures: null,
    lastSeenLocation: "Utown",
    lastSeenTime: new Date(2023, 4, 20, 10, 53, 0, 0),
    lastFedTime: null,
    concernStatus: ["New", "Injured"],
    concernDesc: "New and had wounds on body"
}, {
    catID: 3,
    name: "Mitty",
    photoURLs: [require("../../assets/temp-cat.jpg")],
    gender: "F",
    birthYear: 2020,
    sterilised: true,
    keyFeatures: "White with orange patches with a heart-shaped patch on its back, tends to climb up to high places",
    lastSeenLocation: "Science",
    lastSeenTime: new Date(2023, 4, 15, 18, 34, 0, 0),
    lastFedTime: new Date(2023, 4, 15, 18, 34, 0, 0),
    concernStatus: ["Missing"], // The database registers as missing past certain time?
    concernDesc: "Has not been seen in 5 days"
}, {
    catID: 4,
    name: "Bitty",
    photoURLs: [require("../../assets/temp-cat.jpg")],
    gender: "M",
    birthYear: 2022,
    sterilised: true,
    keyFeatures: "Slightly chubby tabby and meows alot",
    lastSeenLocation: "Arts",
    lastSeenTime: new Date(2023, 4, 20, 9, 16, 0, 0),
    lastFedTime: new Date(2023, 4, 19, 20, 47, 0, 0),
    concernStatus: [], // If the array is empty it's healthy
    concernDesc: null
}];

export { cats }