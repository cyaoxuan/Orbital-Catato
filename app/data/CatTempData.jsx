// Temp data to showcase static pages
const cats = [{
    // For Select FlatList
    catID: 0,
    name: "New Cat",
    photoURL: require("../../assets/placeholder.png"),
    gender: null,
    birthYear: null,
    sterilised: null,
    keyFeatures: null,
    lastSeenLocation: null,
    lastSeenTime: null,
    lastFedTime: null,
    concernStatus: null,
    concernDesc: null
},  {
    catID: 1,
    name: "Kitty",
    photoURL: require("../../assets/temp-cat.jpg"), // Will eventually be an array of 
    gender: "M",
    birthYear: 2019,
    sterilised: true,
    keyFeatures: "Grey tabby and loves to sleep on his back",
    lastSeenLocation: "Engineering", // Will eventually be a geohash
    lastSeenTime: new Date(2023, 5, 19, 14, 21, 0, 0), // Will be converted from firebase timestamp
    lastFedTime: new Date(2023, 5, 19, 12, 1, 0, 0), 
    concernStatus: ["Injured"], 
    concernDesc: "Seen limping, possibly right front leg injured"
}, {
    catID: 2,
    name: "Skitty",
    photoURL: require("../../assets/temp-cat.jpg"),
    gender: null,
    birthYear: null,
    sterilised: false,
    keyFeatures: null,
    lastSeenLocation: "Utown",
    lastSeenTime: new Date(2023, 5, 20, 10, 53, 0, 0),
    lastFedTime: null,
    concernStatus: ["New", "Injured"],
    concernDesc: "New and had wounds on body"
}, {
    catID: 3,
    name: "Mitty",
    photoURL: require("../../assets/temp-cat.jpg"),
    gender: "F",
    birthYear: 2020,
    sterilised: true,
    keyFeatures: "White with orange patches with a heart-shaped patch on its back, tends to climb up to high places",
    lastSeenLocation: "Science",
    lastSeenTime: new Date(2023, 5, 15, 18, 34, 0, 0),
    lastFedTime: new Date(2023, 5, 15, 18, 34, 0, 0),
    concernStatus: ["Missing"], // The database registers as missing past certain time?
    concernDesc: "Has not been seen in 5 days"
}, {
    catID: 4,
    name: "Bitty",
    photoURL: require("../../assets/temp-cat.jpg"),
    gender: "M",
    birthYear: 2022,
    sterilised: true,
    keyFeatures: "Slightly chubby tabby and meows alot",
    lastSeenLocation: "Arts",
    lastSeenTime: new Date(2023, 5, 20, 9, 16, 0, 0),
    lastFedTime: new Date(2023, 5, 19, 20, 47, 0, 0),
    concernStatus: [], // If the array is empty it's healthy
    concernDesc: null
}];

export { cats }