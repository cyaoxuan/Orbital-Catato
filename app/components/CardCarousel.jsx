import { View, FlatList } from "react-native";
import { CatCard } from "../components/CatCard";

const boxWidth = 300;

const Carousel = (props) => {
    return (
        <FlatList style={{ height: 300 }}
                horizontal
                snapToAlignment="center"
                decelerationRate="normal"
                showsHorizontalScrollIndicator={false}
                scrollEventThrottle={10}
                snapToInterval={boxWidth}
                contentOffset={{ x: -120, y: 0 }}
                ItemSeparatorComponent={() => <View style={{width: 20}} />}
                
                data={props.cards}
                renderItem={({item, index}) => {
                    return (
                        <View key={index}> 
                            <CatCard 
                                catImage={item.catImage}
                                catName={item.catName}
                                iconName1={item.iconName1}
                                field1={item.field1}
                                info1={item.info1}
                                iconName2={item.iconName2}
                                field2={item.field2}
                                info2={item.info2}
                            />
                        </View>
                        
                    );
                }}
        />

    );
};

export { Carousel }