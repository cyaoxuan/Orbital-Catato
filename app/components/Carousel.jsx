import { FlatList, View } from "react-native";
import { CatCard } from "./CatCard";
import {
    formatLastFedSimple,
    formatLastSeenSimple,
} from "../utils/formatDetails";

// Function to get info1 field on the card
// @param carouselType: "concern" or "fed"
// @param cat: cat object with concernStatus or lastFedTime
// @return formatted string depending on carouselType or "Unknown" if any fields are missing
function getInfo1(carouselType, cat) {
    if (!cat || !carouselType) {
        return "Unknown";
    }

    if (
        carouselType === "concern" &&
        cat.concernStatus &&
        Object.keys(cat.concernStatus).length !== 0
    ) {
        const concerns = [];
        cat.concernStatus.injured && concerns.push("Injured");
        cat.concernStatus.missing && concerns.push("Missing");
        cat.concernStatus.new && concerns.push("New");

        return concerns.join(", ");
    } else if (carouselType === "unfed" && cat.lastFedTime) {
        return formatLastFedSimple(cat.lastFedTime);
    }
    return "Unknown";
}

// Card Carousel used in Dashboard, implemented using FlatList and snaps to each card
// props: cats (array of cat objects), cardWidth (number), carouselType (string), userRole (object of roles), ...card (props for card)
const CardCarousel = ({
    cats,
    cardWidth,
    carouselType,
    userRole,
    navigation,
    ...card
}) => {
    const spaceBetweenCards = 16;

    return (
        <FlatList
            testID="card-carousel"
            style={{ height: (cardWidth * 6) / 5 }}
            horizontal
            snapToAlignment="center"
            decelerationRate="normal"
            showsHorizontalScrollIndicator={false}
            scrollEventThrottle={4}
            snapToInterval={cardWidth + spaceBetweenCards}
            ItemSeparatorComponent={() => (
                <View style={{ width: spaceBetweenCards }} />
            )}
            data={cats}
            keyExtractor={(item, index) => item.catID}
            renderItem={({ item }) => {
                return (
                    <CatCard
                        name={item.name}
                        photoURL={item.photoURLs ? item.photoURLs[0] : null}
                        cardWidth={cardWidth}
                        profileOnPress={() =>
                            navigation.navigate("catalogue", {
                                screen: "CatProfile",
                                initial: false,
                                params: { catID: item.catID },
                            })
                        }
                        {...card}
                        info1={getInfo1(carouselType, item)}
                        info2={
                            item.locationName && item.locationZone && userRole
                                ? formatLastSeenSimple(
                                      userRole.isCaretaker
                                          ? item.locationName
                                          : item.locationZone,
                                      item.lastSeenTime
                                  )
                                : "Unknown"
                        }
                        showFindLocation={item.lastSeenLocation}
                        locationOnPress={() => {
                            navigation.navigate("Map", {
                                catID: item.catID,
                                location: item.lastSeenLocation,
                            });
                        }}
                        userRole={userRole}
                    />
                );
            }}
        />
    );
};

export { getInfo1, CardCarousel };
