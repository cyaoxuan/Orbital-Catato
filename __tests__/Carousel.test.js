import { render, fireEvent } from "@testing-library/react-native";
import { CardCarousel } from "../app/components/Carousel";
import { cats } from "../app/data/CatTempData"

const mockNavigate = jest.fn();

jest.mock("expo-router", () => ({
  useNavigation: () => ({
    navigate: mockNavigate,
  }),
}));

describe("CardCarousel", () => {
  it("renders carousel successfully", () => {
    const { getByTestId } = render(
      <CardCarousel />
    );
    const carousel = getByTestId("card-carousel");
    expect(carousel).toBeDefined();
  });

  it("renders carousel with cat cards correctly", () => {
    const cardWidth = 300;
    const carouselType = "concern";

    const tree = render(
      <CardCarousel
        cats={cats}
        cardWidth={cardWidth}
        carouselType={carouselType}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("navigates to correct cat profile when card is pressed", () => {
    const cardWidth = 200;
    const carouselType = "concern";

    const { getAllByTestId } = render(
      <CardCarousel
        cats={cats}
        cardWidth={cardWidth}
        carouselType={carouselType}
      />
    );

    const catCards = getAllByTestId("card");
    expect(catCards).toHaveLength(cats.length);

    fireEvent.press(catCards[0]);

    expect(mockNavigate).toHaveBeenCalledWith("catalogue", {
      screen: "CatProfile",
      initial: false,
      params: { catID: cats[0].catID },
    });
  });
});