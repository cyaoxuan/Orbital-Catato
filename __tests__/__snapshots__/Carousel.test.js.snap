// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCarousel /> renders carousel with cat cards correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  data={
    [
      {
        "catID": 1,
        "concernStatus": [
          "Injured",
        ],
        "lastFedTime": 2023-05-19T04:01:00.000Z,
        "lastSeenLocation": "Engineering",
        "name": "Kitty",
        "photoURLs": [
          1,
          1,
        ],
      },
      {
        "catID": 2,
        "concernStatus": [
          "New",
          "Injured",
        ],
        "lastFedTime": null,
        "lastSeenLocation": "Utown",
        "lastSeenTime": 2023-05-20T02:53:00.000Z,
        "name": "Skitty",
        "photoURLs": [
          1,
        ],
      },
      {
        "catID": 3,
        "concernStatus": [
          "Missing",
        ],
        "lastFedTime": 2023-05-15T10:34:00.000Z,
        "lastSeenLocation": "Science",
        "lastSeenTime": 2023-05-15T10:34:00.000Z,
        "name": "Mitty",
        "photoURLs": [
          1,
        ],
      },
    ]
  }
  decelerationRate="normal"
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={4}
  showsHorizontalScrollIndicator={false}
  snapToAlignment="center"
  snapToInterval={324}
  stickyHeaderIndices={[]}
  style={
    {
      "height": 300,
    }
  }
  testID="card-carousel"
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(247, 243, 249, 1)",
            "borderRadius": 12,
            "height": 300,
            "margin": 4,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.15,
            "shadowRadius": 3,
            "width": 300,
          }
        }
        testID="card-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(247, 243, 249, 1)",
              "borderRadius": 12,
              "elevation": 1,
              "flex": 1,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            }
          }
          testID="card-container"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "flexShrink": 1,
                },
                undefined,
              ]
            }
            testID="card"
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#eeeeee",
                    "height": 195,
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 12,
                  },
                  {
                    "height": 150,
                    "resizeMode": "cover",
                    "width": 300,
                  },
                ]
              }
            >
              <Image
                accessibilityIgnoresInvertColors={true}
                siblings={
                  [
                    "Card.Cover",
                    "Card.Title",
                    "Card.Content",
                  ]
                }
                source={1}
                style={
                  [
                    {
                      "flex": 1,
                      "height": undefined,
                      "justifyContent": "flex-end",
                      "padding": 16,
                      "width": undefined,
                    },
                    {
                      "borderRadius": 12,
                    },
                  ]
                }
                testID="cover-image"
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingLeft": 16,
                  },
                  {
                    "minHeight": 50,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 28,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 36,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "minHeight": 30,
                          "paddingRight": 16,
                        },
                        {
                          "marginBottom": 2,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Kitty
                </Text>
              </View>
              <View />
            </View>
            <View
              style={
                [
                  {
                    "paddingHorizontal": 16,
                  },
                  {
                    "paddingBottom": 16,
                  },
                  {
                    "paddingBottom": 0,
                    "paddingHorizontal": 4,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              >
                <Text />
                <Text
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 22,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 28,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      {
                        "flex": 1,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Field: 
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 22,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 28,
                        },
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    Injured
                  </Text>
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              >
                <Text />
                <Text
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 22,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 28,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      {
                        "flex": 1,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Field: 
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 22,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 28,
                        },
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    Engineering
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "width": 24,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(247, 243, 249, 1)",
            "borderRadius": 12,
            "height": 300,
            "margin": 4,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.15,
            "shadowRadius": 3,
            "width": 300,
          }
        }
        testID="card-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(247, 243, 249, 1)",
              "borderRadius": 12,
              "elevation": 1,
              "flex": 1,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            }
          }
          testID="card-container"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "flexShrink": 1,
                },
                undefined,
              ]
            }
            testID="card"
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#eeeeee",
                    "height": 195,
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 12,
                  },
                  {
                    "height": 150,
                    "resizeMode": "cover",
                    "width": 300,
                  },
                ]
              }
            >
              <Image
                accessibilityIgnoresInvertColors={true}
                siblings={
                  [
                    "Card.Cover",
                    "Card.Title",
                    "Card.Content",
                  ]
                }
                source={1}
                style={
                  [
                    {
                      "flex": 1,
                      "height": undefined,
                      "justifyContent": "flex-end",
                      "padding": 16,
                      "width": undefined,
                    },
                    {
                      "borderRadius": 12,
                    },
                  ]
                }
                testID="cover-image"
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingLeft": 16,
                  },
                  {
                    "minHeight": 50,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 28,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 36,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "minHeight": 30,
                          "paddingRight": 16,
                        },
                        {
                          "marginBottom": 2,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Skitty
                </Text>
              </View>
              <View />
            </View>
            <View
              style={
                [
                  {
                    "paddingHorizontal": 16,
                  },
                  {
                    "paddingBottom": 16,
                  },
                  {
                    "paddingBottom": 0,
                    "paddingHorizontal": 4,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              >
                <Text />
                <Text
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 22,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 28,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      {
                        "flex": 1,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Field: 
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 22,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 28,
                        },
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    New, Injured
                  </Text>
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              >
                <Text />
                <Text
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 22,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 28,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      {
                        "flex": 1,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Field: 
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 22,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 28,
                        },
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    Utown
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "width": 24,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(247, 243, 249, 1)",
            "borderRadius": 12,
            "height": 300,
            "margin": 4,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.15,
            "shadowRadius": 3,
            "width": 300,
          }
        }
        testID="card-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(247, 243, 249, 1)",
              "borderRadius": 12,
              "elevation": 1,
              "flex": 1,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            }
          }
          testID="card-container"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "flexShrink": 1,
                },
                undefined,
              ]
            }
            testID="card"
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#eeeeee",
                    "height": 195,
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 12,
                  },
                  {
                    "height": 150,
                    "resizeMode": "cover",
                    "width": 300,
                  },
                ]
              }
            >
              <Image
                accessibilityIgnoresInvertColors={true}
                siblings={
                  [
                    "Card.Cover",
                    "Card.Title",
                    "Card.Content",
                  ]
                }
                source={1}
                style={
                  [
                    {
                      "flex": 1,
                      "height": undefined,
                      "justifyContent": "flex-end",
                      "padding": 16,
                      "width": undefined,
                    },
                    {
                      "borderRadius": 12,
                    },
                  ]
                }
                testID="cover-image"
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingLeft": 16,
                  },
                  {
                    "minHeight": 50,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 28,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 36,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "minHeight": 30,
                          "paddingRight": 16,
                        },
                        {
                          "marginBottom": 2,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Mitty
                </Text>
              </View>
              <View />
            </View>
            <View
              style={
                [
                  {
                    "paddingHorizontal": 16,
                  },
                  {
                    "paddingBottom": 16,
                  },
                  {
                    "paddingBottom": 0,
                    "paddingHorizontal": 4,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              >
                <Text />
                <Text
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 22,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 28,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      {
                        "flex": 1,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Field: 
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 22,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 28,
                        },
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    Missing
                  </Text>
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              >
                <Text />
                <Text
                  style={
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 22,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 28,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      {
                        "flex": 1,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Field: 
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 22,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 28,
                        },
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    Science
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
